### PR TITLE
KB: update the fstrim document with windows info

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Add the windows related info for how to check TRIM status

### CODE REVIEW

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 1]
```diff
@@ -27,8 +27,44 @@ A consequence of the IO errors caused by fstrim is that VMs using affected Longh
 
 Although the described system behavior does not affect data integrity, it might induce panic in some users. Consider the guest Kubernetes cluster scenario. In a stuck VM, the etcd service is unavailable. The effects of this failure cascade from the Kubernetes cluster becoming unavailable to services running on the cluster becoming unavailable.
 
+## How to Confirm the FSTRIM is enabled
```
Comment (bk201) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:30]:
```suggestion
## How to Confirm the fstrim is enabled
```

We can be consistent with the previous title.

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 2]
```diff
@@ -60,4 +96,18 @@ You can determine if fstrim is enabled in VMs that use affected Longhorn volumes
...
-    ```
\ No newline at end of file
+    ```
+
+### Windows
+
+Disable the TRIM operation on Windows will mitigate the described risks, you can refer the following command:
+
+- To disable TRIM operation on ReFS v2:
+```
```
Comment (bk201) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:106]: indent the code blocks for consistency. Check the preview: https://deploy-preview-59--harvester-home-preview.netlify.app/kb/the_potential_risk_with_fstrim

**Review by bk201 — APPROVED** [review]

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 3]
```diff
@@ -27,8 +27,44 @@ A consequence of the IO errors caused by fstrim is that VMs using affected Longh
 
 Although the described system behavior does not affect data integrity, it might induce panic in some users. Consider the guest Kubernetes cluster scenario. In a stuck VM, the etcd service is unavailable. The effects of this failure cascade from the Kubernetes cluster becoming unavailable to services running on the cluster becoming unavailable.
 
+## How to Confirm the fstrim is enabled
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:30]:
```suggestion
## How to Check If fstrim Is Enabled
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 4]
```diff
@@ -27,8 +27,44 @@ A consequence of the IO errors caused by fstrim is that VMs using affected Longh
...
+     Loaded: loaded (/lib/systemd/system/fstrim.timer; enabled; vendor preset: enabled)
+     Active: active (waiting) since Mon 2024-03-18 03:40:24 UTC; 1 week 1 day ago
+    Trigger: Mon 2024-04-01 01:00:06 UTC; 5 days left
+   Triggers: ● fstrim.service
+       Docs: man:fstrim
+
+Mar 18 03:40:24 harvester-cluster-01-pool1-49b619f6-tpc4v systemd[1]: Started Discard unused blocks once a week.
+```
+
+If the fstrim.timer service is enabled, means the system will execute fstrim periodically.
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:49]:
```suggestion
When the fstrim.timer service is enabled, the system periodically runs fstrim.
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 5]
```diff
@@ -27,8 +27,44 @@ A consequence of the IO errors caused by fstrim is that VMs using affected Longh
...
+
+### Windows
+
+On Windows, you can check if fstrim is enabled by running the following command:
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:53]:
```suggestion
You can check if fstrim is enabled by running the following command:
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 6]
```diff
@@ -27,8 +27,44 @@ A consequence of the IO errors caused by fstrim is that VMs using affected Longh
...
+
+```
+C:\> fsutil behavior query DisableDeleteNotify
+NTFS DisableDeleteNotify = 0  (Allows TRIM operations to be sent to the storage device)
+ReFS DisableDeleteNotify = 0  (Allows TRIM operations to be sent to the storage device)
+```
+
+The result `DisableDeleteNotify = 0` means TRIM operations are enabled.
+The detailed information can be found in the [Microsoft documentation](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/fsutil-behavior).
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:62]:
```suggestion
`DisableDeleteNotify = 0` indicates that TRIM operations are enabled. For more information, see [fsutil behavior](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/fsutil-behavior) in the Microsoft documentation.
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 7]
```diff
@@ -60,4 +96,18 @@ You can determine if fstrim is enabled in VMs that use affected Longhorn volumes
     AccuracySec=1h
     Persistent=true
     RandomizedDelaySec=6000
-    ```
\ No newline at end of file
+    ```
+
+### Windows
+
+Disable the TRIM operation on Windows will mitigate the described risks, you can refer the following command:
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:103]:
```suggestion
To mitigate the described risks, you can disable TRIM operations using the following commands:
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 8]
```diff
@@ -60,4 +96,18 @@ You can determine if fstrim is enabled in VMs that use affected Longhorn volumes
...
+
+- To disable TRIM operation on ReFS v2:
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:105]:
```suggestion
- ReFS v2
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 9]
```diff
@@ -60,4 +96,18 @@ You can determine if fstrim is enabled in VMs that use affected Longhorn volumes
...
+```
+C:\> fsutil behavior set DisableDeleteNotify ReFS 1
+```
+
+- To disable TRIM operation on NTFS and ReFS v1:
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:110]:
```suggestion
- NTFS and ReFS v1
```

**Review by jillian-maroket — COMMENTED** [review]
@Vicente-Cheng Let me know if you have concerns about the changes.

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Comment by Vicente-Cheng** [comment]
Hi @bk201, @jillian-maroket,

Sorry, I need your help to review this again.
`fstrim` is usually used on Linux systems because fstrim is the filesystem trim related command.

On the Windows system, we usually call trim or unmap.
So I do some updates to replace the fstrim with the filesystem trim.
Please help to recheck it, thanks!

**Review by bk201 — APPROVED** [review]

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 10]
```diff
@@ -16,20 +16,55 @@ This issue is already resolved by Longhorn v1.5.4, v1.6.1, v1.7.0 and later vers
...
 
 
-Using fstrim is a common way to release unused space in a filesystem. However, this utility is known to cause IO errors when used with Longhorn volumes that are rebuilding. For more information about the errors, see the following issues:
+Filesystem trim is a common way to release unused space in a filesystem. However, this operation is known to cause IO errors when used with Longhorn volumes that are rebuilding. For more information about the errors, see the following issues:
 
 - Harvester: [Issue 4793](https://github.com/harvester/harvester/issues/4739)
 - Longhorn: [Issue 7103](https://github.com/longhorn/longhorn/issues/7103)
 
-## Risks Associated with fstrim Usage
+## Risks Associated with Trim Filesystem Usage
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:24]:
```suggestion
## Risks Associated with Filesystem Trim
```

File: kb/2024-01-30/the_potential_risk_with_fstrim.md [hunk 11]
```diff
@@ -16,20 +16,55 @@ This issue is already resolved by Longhorn v1.5.4, v1.6.1, v1.7.0 and later vers
...
 
-A consequence of the IO errors caused by fstrim is that VMs using affected Longhorn volumes become stuck. Imagine the VM is running critical applications, then becomes unavailable. This is significant because Harvester typically uses Longhorn volumes as VM disks. The IO errors will cause VMs to flap between running and paused states until volume rebuilding is completed.
+A consequence of the IO errors caused by filesystem trim is that VMs using affected Longhorn volumes become stuck. Imagine the VM is running critical applications, then becomes unavailable. This is significant because Harvester typically uses Longhorn volumes as VM disks. The IO errors will cause VMs to flap between running and paused states until volume rebuilding is completed.
 
 Although the described system behavior does not affect data integrity, it might induce panic in some users. Consider the guest Kubernetes cluster scenario. In a stuck VM, the etcd service is unavailable. The effects of this failure cascade from the Kubernetes cluster becoming unavailable to services running on the cluster becoming unavailable.
 
+## How to Check If Trim Filesystem Is Enabled
```
Comment (jillian-maroket) [inline kb/2024-01-30/the_potential_risk_with_fstrim.md:30]:
```suggestion
## How to Check If Filesystem Trim Is Enabled
```

**Review by jillian-maroket — APPROVED** [review]
@Vicente-Cheng Most of the changes are fine. Just make sure that you use "filesystem trim" consistently. Check my comments.

**Pull request merged**